File: ColocKitchenRace/Models/DietaryPreference.swift
//
//  DietaryPreference.swift
//  ColocKitchenRace
//
//  Created by Julien Rahier on 07/02/2026.
//

import Foundation

enum DietaryPreference: String, Codable, CaseIterable, Identifiable, Hashable {
    case vegetarian = "vegetarian"
    case vegan = "vegan"
    case glutenFree = "gluten_free"
    case lactoseFree = "lactose_free"
    case nutFree = "nut_free"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .vegetarian: return "Végétarien"
        case .vegan: return "Végan"
        case .glutenFree: return "Sans gluten"
        case .lactoseFree: return "Sans lactose"
        case .nutFree: return "Sans noix"
        }
    }

    var icon: String {
        switch self {
        case .vegetarian: return "🥬"
        case .vegan: return "🌱"
        case .glutenFree: return "🌾"
        case .lactoseFree: return "🥛"
        case .nutFree: return "🥜"
        }
    }
}

File: android/app/src/main/java/dev/rahier/colockitchenrace/data/model/DietaryPreference.kt
package dev.rahier.colockitchenrace.data.model

enum class DietaryPreference(val displayName: String, val icon: String) {
    VEGETARIAN("Vegetarien", "\uD83E\uDD66"),
    VEGAN("Vegan", "\uD83C\uDF31"),
    GLUTEN_FREE("Sans gluten", "\uD83C\uDF3E"),
    LACTOSE_FREE("Sans lactose", "\uD83E\uDD5B"),
    NUT_FREE("Sans noix", "\uD83E\uDD5C");

    companion object {
        fun fromFirestore(value: String): DietaryPreference? = when (value) {
            "vegetarian" -> VEGETARIAN
            "vegan" -> VEGAN
            "glutenFree" -> GLUTEN_FREE
            "lactoseFree" -> LACTOSE_FREE
            "nutFree" -> NUT_FREE
            else -> null
        }
    }

    fun toFirestore(): String = when (this) {
        VEGETARIAN -> "vegetarian"
        VEGAN -> "vegan"
        GLUTEN_FREE -> "glutenFree"
        LACTOSE_FREE -> "lactoseFree"
        NUT_FREE -> "nutFree"
    }
}
